lib/ansible/modules/web_infrastructure/ansible_tower/tower_host.py

```diff
@@ -82,46 +96,74 @@ def main():
         inventory=dict(required=True),
         enabled=dict(type='bool', default=True),
         variables=dict(),
+        merge_variables=dict(required=False, default=False, type='bool'),
+        groups=dict(required=False, default=[], type='list'),
         state=dict(choices=['present', 'absent'], default='present'),
     )
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
     name = module.params.get('name')
-    description = module.params.get('description')
     inventory = module.params.get('inventory')
-    enabled = module.params.get('enabled')
-    state = module.params.get('state')
+    state = module.params.pop('state')
+    merge_variables = module.params.pop('merge_variables')
+    groups = module.params.pop('groups')
 
     variables = module.params.get('variables')
-    if variables:
-        if variables.startswith('@'):
-            filename = os.path.expanduser(variables[1:])
-            with open(filename, 'r') as f:
-                variables = f.read()
 
     json_output = {'host': name, 'state': state}
 
     tower_auth = tower_auth_config(module)
     with settings.runtime_values(**tower_auth):
         tower_check_mode(module)
         host = tower_cli.get_resource('host')
+        group_res = tower_cli.get_resource('group')
+        params = module.params.copy()
+        params['create_on_missing'] = True
 
         try:
             inv_res = tower_cli.get_resource('inventory')
             inv = inv_res.get(name=inventory)
+            params['inventory'] = inv['id']
+        except exc.notFound as excinfo:
+            module.fail_json(msg='Failed to update the host, inventory not found: {0}'.format(excinfo), changed=False)
+
+        group_ids = []
+        try:
+            for group in groups:
+                group_ids.append(group_res.get(name=group, inventory=params['inventory'])['id'])
+        except exc.NotFound as excinfo:
+            module.fail_json(msg='Group {0} does not exist: {1}'.format(group, excinfo), changed=False)
 
+        try:
+            existing_host = host.get(name=name, inventory=params['inventory'])
+        except exc.NotFound:
+            existing_host = None
+            json_output['created'] = True
+            module.log("Existing host not found, will create")
```

Comment:
There was some other recent issue about the `state: absent` task not working as expected if it had already been deleted. Most of this pertaining to `existing_host` seems like it should either not be done, or be handled specially if the intended state is absent.

That's also something which could be easily added to the test playbook, just copy and paste the task that deletes the host/group and assert that it is not changed.